drivers/drivers.go

```diff
@@ -14,62 +13,57 @@ import (
 // driver represent different ways hosts can be created (e.g. different
 // hypervisors, different cloud providers)
 type Driver interface {
+	// Create a host using the driver's config
+	Create() error
+
 	// DriverName returns the name of the driver as it is registered
 	DriverName() string
 
-	// SetConfigFromFlags configures the driver with the object that was returned
-	// by RegisterCreateFlags
-	SetConfigFromFlags(flags DriverOptions) error
+	// GetIP returns an IP or hostname that this host is available at
+	// e.g. 1.2.3.4 or docker-host-d60b70a14d3a.cloudapp.net
+	GetIP() (string, error)
+
+	// GetSSHHostname returns hostname for use with ssh
+	GetSSHHostname() (string, error)
```

Comment:
I'd convert this to `GetSSHAddr(ess)` it's rare we actually use hostnames, and addr is generic.